mux_test.go

```diff
@@ -449,6 +459,36 @@ func TestPath(t *testing.T) {
 			pathTemplate: `/{category:a|b/c}/{product}/{id:[0-9]+}`,
 			shouldMatch:  true,
 		},
+		{
+			title:        "Path route with regexp alias patterns",
+			route:        new(Route).RegisterPattern("digits", "[0-9]+").Path("/{id:digits}"),
+			request:      newRequest("GET", "http://localhost/1"),
+			vars:         map[string]string{"id": "1"},
+			host:         "",
+			path:         "/1",
+			pathTemplate: `/{id:digits}`,
+			shouldMatch:  true,
+		},
+		{
+			title:        "Path route with regexp alias patterns",
+			route:        new(Route).RegisterPattern("uuid", "[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}").Path("/{category:uuid}/{product:uuid}"),
+			request:      newRequest("GET", "http://localhost/dce51145-5cc3-4b54-bfb0-7bdb64a67e4d/a385ddcb-278e-4234-93dd-4d7b0fcb95c1"),
+			vars:         map[string]string{"category": "dce51145-5cc3-4b54-bfb0-7bdb64a67e4d", "product": "a385ddcb-278e-4234-93dd-4d7b0fcb95c1"},
+			host:         "",
+			path:         "/dce51145-5cc3-4b54-bfb0-7bdb64a67e4d/a385ddcb-278e-4234-93dd-4d7b0fcb95c1",
+			pathTemplate: `/{category:uuid}/{product:uuid}`,
+			shouldMatch:  true,
+		},
+		{
+			title:        "Path route with regexp alias patterns passed through router",
+			route:        NewRouter().RegisterPattern("digits", "[0-9]+").Path("/{id:digits}"),
```

Comment:
Both the test cases look similar to me.
1. First one:
```javascript
{
	title:        "Path route with regexp alias patterns",
	route:        new(Route).RegisterPattern("digits", "[0-9]+").Path("/{id:digits}"),
	request:      newRequest("GET", "http://localhost/1"),
	vars:         map[string]string{"id": "1"},
	host:         "",
	path:         "/1",
	pathTemplate: `/{id:digits}`,
	shouldMatch:  true,
}
```

2. Second:
```javascript
{
	title:        "Path route with regexp alias patterns passed through router",
	route:        NewRouter().RegisterPattern("digits", "[0-9]+").Path("/{id:digits}"),
	request:      newRequest("GET", "http://localhost/1"),
	vars:         map[string]string{"id": "1"},
	host:         "",
	path:         "/1",
	pathTemplate: `/{id:digits}`,
	shouldMatch:  true,
}
```

We should add at least one negative test case. What do you think?

Reply:
Yeah, seems like this tests was very similar, deleted second test and added negative test case